### PR TITLE
etl/type_traits.h: Add support for type_identity

### PR DESCRIPTION
**Pull request by tigran2008**

### CODE REVIEW

**Comment by jwellbelove** [comment]
Did you add a unit test?

**Comment by tigran2008** [comment]
Sorry, I overlooked that part :/

I'm away from my computer right now, but would the following be enough if I added it later?

Given this function:

```cpp
template <typename T>
T add(T first, typename etl::type_identity<T>::type second) {
    return first + second;
}
```

The result of `add(2.5, 1)` would be checked if it was equal to `3.5`

**Comment by jwellbelove** [comment]
That looks OK.

File: test/test_type_traits.cpp [hunk 1]
```diff
@@ -1320,4 +1327,9 @@ namespace
     CHECK_FALSE(bool(etl::is_base_of_all<Base, D1, D2, D3, D4>::value));
 #endif
   }
+
+  //*************************************************************************
+  TEST(test_type_identity) {
+    CHECK(type_identity_test_add(1.5f, 2) == 3.5f);
```
Comment (jwellbelove) [inline test/test_type_traits.cpp]: You should never test floating point numbers for equality.
Use `CHECK_CLOSE` with a small tolerance value, such as `0.01f`.

Comment (tigran2008) [reply test/test_type_traits.cpp]: Why not? I used 0.5 as the fraction, which is a negative power of two, so no rounding is involved.

**Comment by tigran2008** [comment]
Well, I changed the CHECK into a CHECK_CLOSE.

File: test/test_type_traits.cpp (hunk 1 above)
Comment (jwellbelove) [reply test/test_type_traits.cpp]: It's just standard good practice.
It protects against code breaking if someone were to change a value in the future without realising the implications.

**Pull request merged**